library/src/main/java/com/novoda/downloadmanager/MigrationFactory.java

```diff
@@ -0,0 +1,29 @@
+package com.novoda.downloadmanager;
+
+import android.content.Context;
+import android.database.sqlite.SQLiteDatabase;
+
+import java.io.File;
+
+public final class MigrationFactory {
+
+    private MigrationFactory() {
+        // Uses static methods.
+    }
+
+    public static Migrator createVersionOneToVersionTwoMigrator(Context context, File databasePath, Migrator.Callback callback) {
+        if (!databasePath.exists()) {
+            return Migrator.NO_OP;
+        }
+
+        SQLiteDatabase sqLiteDatabase = SQLiteDatabase.openDatabase(databasePath.getAbsolutePath(), null, 0);
+        SqlDatabaseWrapper database = new SqlDatabaseWrapper(sqLiteDatabase);
```

Comment:
Seems a bit off having logic and DB opening in a factory class. Functionality creeping into a factory

Reply:
I don't think this is logic per se, we just have `SqlDatabaseWrapper` covering the Android class and exposing a few methods on it, for test purposes. (Don't mock types you don't own, etc.) We could perhaps move the `openDatabase` call to the wrapper though? would that be better?

Reply:
Sorry I meant the logic as in the branching just above, kinda combined two comments.

As for the DB opening, it seems to me the same as putting too much in constructors. If I call a factory I wouldn't really expect it to fail, and I might be surprised it was opening a DB unless the thing I was constructing directly was a "DB" object.